src/Strategy/Go/GoListPackages.hs

```diff
@@ -0,0 +1,240 @@
+-- The end result is a graph of module dependencies for only modules that have at least one of its packages used by the built project.
+--
+-- Other features of this function:
+-- * Removes standard lib deps where they are called out by a flag.
+--   Stdlib packages with pseudo versions are not filtered yet. This is the same behavior as the current dynamic tactic in Strategy.Go.GoList.
+-- * Removes path dependencies and child deps that aren't used elsewhere in the graph.
+--   The go tools give us this data but we haven't decided yet how to present it.
+-- * Replaces modules according to the module's 'replacement' field.
+buildGraph :: (Has Diagnostics sig m) => [GoPackage] -> m (Graphing.Graphing Dependency, GraphBreadth)
+buildGraph rawPackages =
+  fmap ((,Complete) . pruneUnreachable) -- Remove deps that are only children of removed path deps.
+    . evalGrapher
+    . traverse_ graphEdges
+    $ pkgsNoStdLibImports
+  where
+    (stdLibImportPaths, pkgsNoStdLibImports) = foldl' go (HashSet.empty, HashMap.empty) rawPackages
+
+    go :: (HashSet.HashSet ImportPath, HashMap.HashMap ImportPath GoPackage) -> GoPackage -> (HashSet.HashSet ImportPath, HashMap.HashMap ImportPath GoPackage)
+    go (stdLibPaths, otherPackages) g@GoPackage{standard, importPath}
+      | standard = (HashSet.insert importPath stdLibPaths, otherPackages)
+      | otherwise = (stdLibPaths, HashMap.insert importPath (removeIgnoredPackages g) otherPackages)
+
+    -- "C" is a special package for using Go's FFI.
+    -- Ignore it because trying to look it up as a package import path later will fail.
+    ignoredPackages :: HashSet.HashSet ImportPath
+    ignoredPackages = HashSet.insert (ImportPath "C") stdLibImportPaths
+
+    removeIgnoredPackages :: GoPackage -> GoPackage
+    removeIgnoredPackages g@GoPackage{packageDeps = pDeps} =
+      g{packageDeps = filter (\i -> not $ i `HashSet.member` ignoredPackages) pDeps}
+
+    graphEdges :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => GoPackage -> m ()
+    graphEdges GoPackage{importPath, packageDeps, listError} = do
+      maybe (pure ()) warn listError
+      currModule@GoModule{isMainModule, indirect} <- importToModule importPath
+      unless (isMainModule || isPathDep currModule) $ do
+        let currDep :: Dependency
+            currDep = modToDep currModule
+
+            addChildEdge :: (Has Diagnostics sig m, Has (Grapher Dependency) sig m) => Dependency -> m ()
+            addChildEdge childDep = do
+              when (childDep /= currDep) $
+                edge currDep childDep
+
+            makeEdge :: (Has (Grapher Dependency) sig m, Has Diagnostics sig m) => ImportPath -> m ()
+            makeEdge = importToModule >=> pure . modToDep >=> addChildEdge
+        traverse_ makeEdge packageDeps
+        if indirect
+          then deep currDep
```

Comment:
Whats the purpose of this `deep` function? Why do we have to explicitly set the dep to be deep? Isn't knowing if it is direct or not enough?